amethyst_core/src/named.rs

```diff
@@ -0,0 +1,37 @@
+use specs::{Component, DenseVecStorage, EntityBuilder, WriteStorage};
+
+/// A component that gives a name to an `Entity`
+pub struct Named {
+    pub name: &'static str,
+}
+
+impl Named {
+    pub fn new(name: &'static str) -> Self {
+        Named { name }
+    }
+}
+
+impl Component for Named {
+    type Storage = DenseVecStorage<Self>;
+}
+
+/// An easy way to name an `Entity` and give it a `Named` `Component`.
+pub trait WithNamed
+where
+    Self: Sized,
+{
+    /// Adds a name to the entity being built.
+    fn named(self, name: &'static str) -> Self;
+}
+
+impl<'a> WithNamed for EntityBuilder<'a> {
```

Comment:
Psst, add one for the other builder types too

Reply:
which are?

Reply:
[Lazy](https://docs.rs/specs/0.12.1/specs/world/struct.LazyBuilder.html) and [EntityRes](https://docs.rs/specs/0.12.1/specs/world/struct.EntityResBuilder.html)

Reply:
Added LazyBuilder. I cannot add EntityResBuilder without changing the signature of the interface, so I'm not sure what to do about that.

Reply:
You could add a second trait with the new interface

Reply:
I'm not sure if it makes more sense to have a second interface for the same thing, or to just let people use 
```rs
.with(Named::new("my name"), &mut named_storage)
```
vs
```rs
use amethyst::ecs::world::EntityResBuilder;
.named("my name", &mut named_storage)
```

Reply:
Yeah I'll leave that choice to you, just wanted to clue you in to the existence of the other builders.

Reply:
I'll go with only "with" for now. I feel like having two traits would be more confusing than anything.

Reply:
Thanks for teaching me :)